### PR TITLE
[rn-swipeable-panel] Remove, bundles its own types

### PR DESCRIPTION
**Pull request by BendingBender**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: notNeededPackages.json [hunk 1]
```diff
@@ -3742,6 +3742,10 @@
             "libraryName": "riot",
             "asOfVersion": "4.1.0"
         },
+        "rn-swipeable-panel": {
+            "libraryName": "rn-swipeable-panel",
+            "asOfVersion": "1.2.2"
```
Comment (ImRodry) [inline notNeededPackages.json:3747]: Couldn't find any proof of this but types first started appearing on v1.2.1 (tried by installing the package myself)
```suggestion
            "asOfVersion": "1.2.1"
```

**Review by ImRodry — APPROVED** [review]

File: notNeededPackages.json (hunk 1 above)
Comment (BendingBender) [reply notNeededPackages.json:3747]: Not possible, there is already `@types/rn-swipeable-panel@1.2.1` published on npm. I simply took the next version.

Comment (ImRodry) [reply notNeededPackages.json:3747]: This is supposed to be the number of the version that added type declarations for this package, it's not the version of the `@types` package

Already told you this twice on two other PRs, please...

**Review by ImRodry — CHANGES_REQUESTED** [review]
Accidentally pressed approve

File: notNeededPackages.json (hunk 1 above)
Comment (BendingBender) [reply notNeededPackages.json:3747]: The checks won't pass. You can try it out yourself with npm run test-all. This is actually supposed to be the version of the @types package that will be published with the obsoleted warning.

Comment (ImRodry) [reply notNeededPackages.json:3747]: Yeah my bad

**Review by ImRodry — APPROVED** [review]

**Pull request merged**